Jump/resource/json/config.txt
{

    "mainConfig:{

        "fps": 60,帧率
    },

    "stageConfig":[

        第一个表示无尽的配置
        {
            "maxLevel": 20,终点高度，填写0表示无尽

            "heightAddSpeed": 0.625,摄像机上升速度

            "coinPropProbability": 0.3,金币出现概率

            "maxCoinNum": 3,金币上限

            "coinXSpeed": -1,金币x轴移动速度

            "coinJumpHeight": 2.4,金币弹跳高度

            "itemPropProbability": 0.3,物品出现概率

            "maxItemNum": 3,最大物品数量

            "itemXSpeed": -1,物品x轴移动速度

            "itemJumpHeight": 2.4,物品弹跳高度

            "enemyPropProbability":  0.2,敌人出现概率

            "maxEnemyNum":  2,最大敌人是数量

            "linePropProbability": 0.2,线出现的概率

            "maxLineNum": 1,线的数量上限
        },

        之后每一个都是挑战的一关
        {
            "maxLevel": 20,终点高度，填写0表示无尽

            "heightAddSpeed": 0.625,摄像机上升速度

            "coinPropProbability": 0.3,金币出现概率

            "maxCoinNum": 3,金币上限

            "coinXSpeed": -1,金币x轴移动速度

            "coinJumpHeight": 2.4,金币弹跳高度

            "itemPropProbability": 0.3,物品出现概率

            "maxItemNum": 3,最大物品数量

            "itemXSpeed": -1,物品x轴移动速度

            "itemJumpHeight": 2.4,物品弹跳高度

            "enemyPropProbability":  0.2,敌人出现概率

            "maxEnemyNum":  2,最大敌人是数量

            "linePropProbability": 0.2,线出现的概率

            "maxLineNum": 1,线的数量上限
        }
    ],

    "gameConfig":{

        "ladderWidthFix": 10,最高一层楼梯水平宽度

        "physicalTimeFix": 1.3,物理引擎时间系数(废弃的参数，如果修改会关联到所有的力，所以先不要去改动他了)

        "worldTimeFix": 1,世界时间缩放系数

        "factor":  80,画面缩放系数

        "unitHeight":  2.4,台阶高度

        "unitWidth":  2,台阶宽度

        "triangleWidth":  0.3,台阶切角宽度

        "triangleHeight":  0.4,台阶切角高度

        "unitNum":  15,台阶总数

        "changeUnitNum":  5,移动多少台阶后重置位置

        "humanLength":  0.7,人的长度

        "humanRadius":  0.3,人的半径

        "jumpAngle":  0.935,跳跃角度

        "jumpForce": [230,450],跳跃时受到的力

        public jumpPoint:number[] = [0,0],跳跃受力点

        "jumpDisableTime":  300,跳跃后禁止跳跃时间

        "friction":  0,阻力系数

        "relaxation":  10,缓和系数

        "restitution": 0,弹力系数

        "humanDamping": 0.1,人的空气阻力

        "humanAngularDamping": 0.1,人的角空气阻力

        "humanSleepXFix":  0.5,人的移动修正

        "humanGravityScale": 1,人的重力系数

        "enemyJumpProbability":  1,敌人跳跃概率

        "propHeightFix":  10,敌人、金币、线出现位置修正

        "cameraFollowSpeedFix":  0.02,摄像机跟随系数

        "firstCameraFollowTime": 2000,初始摄像机修正时间

        "physicsEngineFps": 60,物理引擎的帧率

        "humanStartPos": [-4,9],出生位置

        "firstJumpAngle": 0.7,第一次跳跃角度

        "firstJumpForce": [400,450],第一次跳跃的力

        "firstJumpPoint": [0,0],第一次跳跃的受力点

        "lineWidth": 0.5,线宽
        
        "lineJumpAngle": 1.2,触线跳跃角度

        "lineJumpForce": [400,700],触线跳跃力度

        "lineJumpPoint": [0,0],触线跳跃力受力点

        "coinRadius": 0.15,金币半径

        "coinMoveToHumanSpeed": 8,金币追逐玩家的速度

        "coinMoveToHumanAngularSpeed": 0.2,金币追逐玩家的角速度

        "coinMoveToHumanRadius": 5,金币追逐玩家的范围半径

        "gravity": [0, -9.78],重力系数

        "humanFixForce": [0,-0.3],人物修正力

        "humanFixForcePoint": [0.35,0]，人物修正力的受力点

        "solverIterations": 1,最大求解次数

        "itemRadius":  0.5,物品半径

        "humanBigSize": 2,人物放大系数

        "humanBigMassFix": 2,人物放大的质量系数

        "humanBigJumpForceFix": 2,人物放大的弹跳受力系数

        "humanFeatherMassFix": 0.5,人物变轻的质量系数

        "humanFeatherJumpForceFix": 0.5,人物变轻的弹跳受力系数

        "humanSlowFix": 0.5,速度放慢的系数

        "coinDoubleFix": 2,金币加成的系数

        "itemBtTime": 10000,获得物品后按钮持续时间

        "itemEffectTime": 10000,使用物品后持续时间

        "redLineMoneyChange": -10,踩红线跳金钱变化数量

        "greenLineMoneyChange": 10,踩绿线跳金钱变化数量

        "coinMoneyChange": 1,吃金币金钱变化数量

        "collisionCheckFix": 0,包围盒检测的修正系数

        "rewardGravity": 9.8,金币和物品的重力系数
    }
}

Jump/src/Config.ts
class Config{

    public gameConfig:GameConfig;

    public mainConfig:MainConfig;

    public stageConfig:StageConfig[];
}

class MainConfig{

    public fps:number = 60;
}

class StageConfig{

    public maxLevel:number = 0;

    public heightAddSpeed:number = 0.625;

    public coinPropProbability:number = 0.3;

    public maxCoinNum:number = 3;

    public coinXSpeed:number = -1;

    public coinJumpHeight:number = 2.4;

    public itemPropProbability:number = 0.3;

    public maxItemNum:number = 3;

    public itemXSpeed:number = -1;

    public itemJumpHeight:number = 2.4;

    public enemyPropProbability:number = 0.2;

    public maxEnemyNum:number = 2;

    public linePropProbability:number = 0.2;

    public maxLineNum:number = 1;
}

class GameConfig{

    public ladderWidthFix:number = 10;

    public ladderHeightFix:number = 10;

    public ladderXFix:number = 1;

    public finalLadderXFix:number = 2;

    public terminalWidth:number = 5;

    public physicalTimeFix:number = 1.3;

    public worldTimeFix:number = 1;

    public factor:number = 80;

    public unitHeight:number = 2.4;

    public unitWidth:number = 2;

    public triangleWidth:number = 0.3;

    public triangleHeight:number = 0.4;

    public unitNum:number = 20;

    public ladderChangeUnitNum:number = 1;

    public ladderChangeUnitFix:number = 2;

    public humanLength:number = 0.7;

    public humanRadius:number = 0.3;

    public jumpAngle:number = Math.PI * 0.25 + 0.15;

    public jumpForce:number[] = [80,145];

    public jumpPoint:number[] = [0,0];

    public jumpDisableTime:number = 300;

    public friction:number = 0;

    public relaxation:number = 10;

    public restitution:number = 0;

    public humanDamping:number = 0.1;

    public humanAngularDamping:number = 0.1;

    public humanGravityScale:number = 1;

    public humanSleepXFix:number = -0.5;

    public enemyJumpProbability:number = 1;

    public propHeightFix:number = 10;

    public finalPropHeightFix:number = 5;

    public cameraFollowSpeedFix:number = 0.02;

    public firstCameraFollowTime:number = 1000;

    public physicsEngineFps:number = 30;

    public humanStartPos:number[][] = [[100,100]];

    public firstJumpAngle:number = Math.PI * 0.25 + 0.15;

    public firstJumpForce:number[] = [100, 100];

    public firstJumpPoint:number[] = [0,0];

    public lineWidth:number = 0.5;

    public lineJumpAngle:number = 1;

    public lineJumpForce:number[] = [400,600];

    public lineJumpPoint:number[] = [0,0];

    public coinRadius:number = 0.3;

    public coinMoveToHumanSpeed:number = 8;

    public coinMoveToHumanAngularSpeed:number = 0.2;

    public coinMoveToHumanRadius:number = 5;

    public gravity:number[] = [0, -9.78];

    public humanFixForce:number[] = [0,-0.3];

    public humanFixForcePoint:number[] = [0.35,0];

    public solverIterations:number = 1;

    public itemRadius:number = 0.5;

    public humanBigSize:number = 2;

    public humanBigMassFix:number = 2;

    public humanBigJumpForceFix:number = 2;

    public humanFeatherMassFix:number = 0.5;

    public humanFeatherJumpForceFix:number = 0.5;

    public humanSlowFix:number = 0.5;

    public coinDoubleFix:number = 2;

    public itemBtTime:number = 5000;

    public itemEffectTime:number = 5000;

    public redLineMoneyChange:number = -10;

    public greenLineMoneyChange:number = 10;

    public coinMoneyChange:number = 1;

    public collisionCheckFix:number = 0.5;

    public rewardGravity:number = 9.8;
}